code/controllers/subsystem/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -949,3 +949,41 @@
 
 #undef MALF_ION_PROB
 #undef REPLACE_LAW_WITH_ION_PROB
+
+/// Midround Voidwalker Ruleset (From Ghosts)
+/datum/dynamic_ruleset/midround/from_ghosts/voidwalker
+	name = "Voidwalker"
+	midround_ruleset_style = MIDROUND_RULESET_STYLE_LIGHT
+	antag_datum = /datum/antagonist/voidwalker
+	antag_flag = ROLE_VOIDWALKER
+	antag_flag_override = ROLE_VOIDWALKER
+	ruleset_category = parent_type::ruleset_category |  RULESET_CATEGORY_NO_WITTING_CREW_ANTAGONISTS
+	required_enemies = list(2,2,1,1,1,1,1,0,0,0)
+	required_candidates = 1
+	weight = 2
+	cost = 5
+	minimum_players = 40
+	repeatable = TRUE
+	ruleset_lazy_templates = list(LAZY_TEMPLATE_KEY_VOIDWALKER_VOID)
+
+/datum/dynamic_ruleset/midround/from_ghosts/voidwalker/acceptable(population = 0, threat_level = 0)
+	var/turf/spawn_loc = find_space_spawn()
+	if(!spawn_loc)
+		return FALSE
+	return ..()
+
+/datum/dynamic_ruleset/midround/from_ghosts/voidwalker/generate_ruleset_body(mob/applicant)
+	var/datum/mind/player_mind = new /datum/mind(applicant.key)
+	player_mind.active = TRUE
+
+	var/mob/living/carbon/human/voidwalker = new (find_space_spawn())
+	player_mind.transfer_to(voidwalker)
+	player_mind.set_assigned_role(SSjob.GetJobType(/datum/job/voidwalker))
+	player_mind.special_role = antag_flag
+	player_mind.add_antag_datum(antag_datum)
+	voidwalker.set_species(/datum/species/voidwalker)
```

Comment:
Any reason not to make a `human/species/voidwalker` subtype for ease of spawning? Admins might appreciate it.